src/features/filter.ts

```diff
@@ -1,12 +1,20 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { PayloadAction, Slice, createSlice } from '@reduxjs/toolkit';
+import { Status } from '../types/Status';
 
 const initialState = {
   query: '',
   status: 'all',
 };
 
-export const filterSlice = createSlice({
+export const filterSlice: Slice = createSlice({
   name: 'filter',
   initialState,
-  reducers: {},
+  reducers: {
+    setFilter: (state, { payload }: PayloadAction<Status>) => {
+      return { ...state, status: payload };
```

Comment:
By the way, Redux Toolkit's createSlice uses Immer under the hood, which allows you to mutate the state directly without needing to return a new object.
```suggestion
      state.query = action.payload;
```